lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -340,7 +340,8 @@ struct StoreOpConversion
           elem = bitcast(elem, valueElemTy);
 #ifdef USE_ROCM
           Value maskVal = llMask ? maskElems[vecStart] : int_val(1, 1);
-          Value ret = select(maskVal, elem , bitcast(i32_val(0), valueElemTy));
+          Value ret = select(maskVal, elem,
+                             bitcast(int_val(valueElemNbits, 0), valueElemTy));
```

Comment:
@dfukalov `int_val(valueElemNbit, 0)` is still incorrect. What we really need is to conditionally emit `store` when `maskVal` is non-zero. But for the purpose of this PR I think it's fine. @weihanmines is already working on a solution now.

Reply:
Yes, writing zero is incorrect, but creating incorrect bitcast just fails the compilation. Such a (corrected) bitcast anyway might be needed in masked load.